src/commands/AddCommand.js

```diff
@@ -0,0 +1,222 @@
+        if (!addable) {
+          this.logger.warn(`Will not add ${filteredPackage.name} to itself.`);
+        }
+        return addable;
+      })
+      // Skip packages without actual changes to manifest
+      .filter(filteredPackage => {
+        const deps = filteredPackage[this.dependencyType] || {};
+
+        // Check if one of the packages to install necessiates a change to filteredPackage's manifest
+        return this.packagesToInstall
+          .filter(pkgToInstall => pkgToInstall.name !== filteredPackage.name)
+          .some(pkgToInstall => {
+            if (!(pkgToInstall.name in deps)) {
+              return true;
+            }
+            const current = deps[pkgToInstall.name];
+            const range = getRangeToReference(current, pkgToInstall.version, pkgToInstall.versionRange);
+            return range !== current;
+          });
+      });
+
+
+    if (this.packagesToChange.length === 0) {
+      const packagesToInstallList = this.packagesToInstall.map(pkg => pkg.name).join(', ');
+      this.logger.warn(`No packages found in scope where ${packagesToInstallList} can be added.`);
+    }
+
+    callback(null, this.packagesToChange.length > 0);
+  }
+
+  execute(callback) {
+    this.logger.info(`Add ${this.dependencyType} in ${this.packagesToChange.length} packages`);
+
+    this.packagesToChange
+      .forEach(pkgToChange => {
+        const manifestPath = path.join(pkgToChange.location, "package.json");
+
+        const notSamePackage = pkgToInstall => pkgToChange.name !== pkgToInstall.name;
+
+        const applicable = this.packagesToInstall
+          .filter(notSamePackage)
+          .reduce((results, pkgToInstall) => {
+            const deps = pkgToChange[this.dependencyType] || {};
+            const current = deps[pkgToInstall.name];
+            const range = getRangeToReference(
+              current,
+              pkgToInstall.version,
+              pkgToInstall.versionRange
+            );
+            const id = `${pkgToInstall.name}@${range}`;
+            const message = `Add ${id} as ${this.dependencyType} in ${pkgToChange.name}`;
+            this.logger.verbose(message);
+            results[pkgToInstall.name] = range;
+            return results;
+          }, {});
+
+        const source = FileSystemUtilities.readFileSync(manifestPath);
+
+        const file = new JSONFile(manifestPath, source);
+        file.set({[this.dependencyType]: applicable});
+
+        const rewritten = file.stringify();
+
+        if (rewritten !== source) {
+          FileSystemUtilities.writeFileSync(manifestPath, rewritten);
+        }
+      });
+
+      this.logger.info(`Changes require bootstrap in ${this.packagesToChange.length} packages`);
+
+      // Bootstrap all changed packages
+      const flags = Object.assign({}, this.flags, {
+        scope: this.packagesToChange.map(pkgToChange => pkgToChange.name)
+      });
+
+      return new BootstrapCommand([], flags, this.cwd).run()
```

Comment:
Ah, I see why the `constructor` is there. Actually, you don't need `this.flags`, you can use `this.options` to merge the updated scope onto. `this.cwd` should be `this.repository.rootPath`, which is equivalent for the purposes of calling another command.